src/TensorFlowNET.Core/Tensors/TensorShape.cs

```diff
@@ -33,7 +33,23 @@ public class TensorShape
         /// <summary>
         ///     Returns the size this shape represents.
         /// </summary>
-        public int size => shape.Size;
+        public int size
```

Comment:
Don't understand why we need this expanded functions? I thought `shape.Size` already computed the size.

Reply:
When shape contains -1 or 0 this will break. TensorShape.size is not something that exists in TF - I've looked into all of the use cases and it is mostly is size is > 0 or > 1